### PR TITLE
GEODE-3539: cleanup GfshCommand and refactor tests

### PR DESCRIPTION
**Pull request by jinmeiliao**
* remove more try catch blocks in commands
* make GfshCommand a pure wrapper around CliUtil for easy mocking

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommandTest.java [hunk 1]
```diff
@@ -144,4 +159,55 @@ private void addJvmOptionsForOutOfMemoryErrors(final List<String> commandLine) {
...
+
+  @Rule
+  public GfshParserRule commandRule = new GfshParserRule();
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
```
Comment (pdxrunner) [inline geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommandTest.java]: Unused rule, should be deleted along with the import for `TemporaryFolder`

File: geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommandTest.java [hunk 2]
```diff
@@ -144,4 +159,55 @@ private void addJvmOptionsForOutOfMemoryErrors(final List<String> commandLine) {
...
+  private StartLocatorCommand spy;
+
+  @Before
+  public void before() throws Exception {
+    spy = Mockito.spy(StartLocatorCommand.class);
+    doReturn(mock(Gfsh.class)).when(spy).getGfsh();
+  }
+
```
Comment (pdxrunner) [inline geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommandTest.java]: I'd prefer to see these new @Rule & @Before methods moved to the top of the file with the original @Before method.

File: geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartServerCommandTest.java [hunk 3]
```diff
@@ -141,4 +156,54 @@ public void testCreateServerCommandLineWithRestAPI() throws Exception {
...
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  private StartServerCommand spy;
+
+  @Before
+  public void before() throws Exception {
+    spy = Mockito.spy(StartServerCommand.class);
+    doReturn(mock(Gfsh.class)).when(spy).getGfsh();
+  }
```
Comment (pdxrunner) [inline geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartServerCommandTest.java]: See above comments on StartLocatorCommandTest

File: geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DestroyRegionCommandTest.java [hunk 4]
```diff
@@ -73,4 +83,35 @@ public void whenNoRegionIsFoundOnAnyMembers() throws Exception {
...
+    when(result2.isSuccessful()).thenReturn(false);
+    when(result2.getMessage()).thenReturn("result2 message");
+
+    result = parser.executeCommandWithInstance(command, "destroy region --name=test");
+    assertThat(result.getStatus()).isEqualTo(Result.Status.ERROR);
+    assertThat(result.getContent().toString()).contains("result2 message");
+
+    // verify that xmlEntiry returned by the result1 is not saved to Cluster config
+    verify(command, never()).persistClusterConfiguration(any(), any());
+  }
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DestroyRegionCommandTest.java]: Could have a couple more tests, this one multiple-results tests looks like it misses coverage of some conditional clauses in the foreach in `DestroyRegionCommand.destroyRegion`: 1) a result contains a throwable 2) a not-successfull result has a null message.

**Comment by jinmeiliao** [comment]
precheckin green

**Pull request closed without merging**